lib/boom_notifier/mail_notifier.ex

```diff
@@ -31,19 +31,32 @@ defmodule BoomNotifier.MailNotifier do
           {:mailer, module()} | {:from, String.t()} | {:to, String.t()} | {:subject, String.t()}
   @type options :: [option]
 
+  @impl BoomNotifier.Notifier
+  def validate!(options) do
+    with _mailer <- Keyword.fetch!(options, :mailer),
+         _from <- Keyword.fetch!(options, :from),
+         _to <- Keyword.fetch!(options, :to),
+         _subject <- Keyword.fetch!(options, :subject) do
+      nil
+    end
```

Comment:
Not fully relevant if you take my previous idea, but overall about Elixir `with`.

Note this is equivalent to:

```ex
def validate!(options) do
  Keyword.fetch!(options, :mailer)
  Keyword.fetch!(options, :from)
  Keyword.fetch!(options, :to)
  Keyword.fetch!(options, :subject)
  nil
end
```

so no major gain from using `with` in this case, IMHO

Reply:
I agree with you but I think Credo complains when using `Keyword.fetch!/1` without assigns but this could also be solved using something like this

```elixir
def validate!(options) do
    _mailer = Keyword.fetch!(options, :mailer)
    _from = Keyword.fetch!(options, :from)
    _to = Keyword.fetch!(options, :to)
    _subject = Keyword.fetch!(options, :subject)
    nil
  end
 ```
 
 Anyways, I think it worths give your previous idea a try.